src/components/UniqueSlug/index.tsx

```diff
@@ -115,16 +142,20 @@ export const UniqueSlug: React.FC<{
       <Text
         label={label}
         initialValue={initialValue}
-        onChange={setValue}
-        showError={Boolean(error || !slugIsValid)}
+        onChange={newSlug => {
+          setValue(newSlug)
+          dispatchSlugValidation({ type: 'SET_SLUG', payload: newSlug })
```

Comment:
Setting the value in two places here is duplicative, `dispatchSlugValidation` should not contain the slug, it has it's own state unless we want to move purely to the reducer pattern